Add template funcs countwords and countrunes

digitalcraftsman opened the pull request:
Takes any type, tries to conver it into a string and returns the total number of words.

bep commented on docs/content/templates/functions.md:
```diff
@@ -460,18 +524,25 @@ e.g.
 * `{{substr "BatMan" 3 3}}` → "Man"
 
+
+***
```
What's all these asterisks?

digitalcraftsman commented:
I revisited the functions and aspects @bep mentioned in his comments. To support CJK languages I added the `countrunes` function.

Both `countwords` and `countrunes` behave like `.WordCount` and `.RuneCount`. In a nutshell, I extracted the code that both page variables use to determine the content length, wrapped them into template functions and tried different text snippets.

Usually all functions have a test case to check if they work properly. Since both template functions use the same code as `.WordCount` and `.RuneCount` I looked at their testing functions.

A [quick look](https://github.com/spf13/hugo/blob/49fe04c0bd8111bf686d9205d543f8651ea24cfc/hugolib/page_test.go#L585) at them showed that they used sample content defined in constants. I used them to test my template functions manually and compared the result with the exptected values in the existing test functions and got the same results. So, should I add testing functions anyway?

Finally, I removed the asterisks that were initially added as spacers between the functions to get a better overview.

digitalcraftsman commented:
Is this PR in a mergable state or is something missing?

bep commented:
It is mergable -- but I wonder: What is the use case for this? When do you need to count words?

digitalcraftsman commented:
I thought especially about external ressources that you can fetch with `getJSON`.

digitalcraftsman commented:
Is there any interest for such template functions or should I close this issue?

/cc @bep

spf13 commented:
@digitalcraftsman @bep I'm fine with this code. I'm not sure what the valid use is for it though. If anyone has one, happy to merge. I'd be surprised if someone doesn't have one for it and it doesn't really add any weight to have this functionality.

bep commented:
I was going to say documentation, but I notice @digitalcraftsman has supplied perfect docs for this, so yes -- I don't need this, but I'm sure someone will. I'm fine with this.

spf13 commented:
merged as dfa34af
